src/components/OfficeLocations/OfficeLocationsComponent.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Form, Icon, Table } from 'semantic-ui-react';
+
+import DropdownComponent from '../common/DropdownComponent';
+
+import generateDropdownOptions from '../../_utils/generateDropdownOptions';
+import verifySuperAdmin from '../../_utils/verifySuperAdmin';
+
+export default class OfficeLocations extends Component {
+  state = {
+    show: false,
+    location: this.props.location
+  };
+
+  componentDidMount() {
+    if (this.props.showFetchLocations) {
+      this.props.loadOfficeLocations();
+    }
+  }
+
+  handleChange = (event, data) => {
+    event.stopPropagation();
+
+    this.setState({
+      location: data.value
+    });
+  };
+
+  toggleFormVisibility = () => {
+    this.setState({
+      show: !this.state.show
+    });
+  };
+
+  updateLocation = () => {
+    const { assetDetail, updateAsset, uuid } = this.props;
+    const { asset_code, model_number, serial_number } = assetDetail;
+
+    updateAsset(uuid, {
+      asset_location: this.state.location,
+      model_number,
+      asset_code,
+      serial_number
+    }).then(() => {
+      this.toggleFormVisibility();
+    });
+  };
+
+  render() {
+    const { assetDetail, updateLoading, locationLoading, locationList } = this.props;
+    const { location, show } = this.state;
+
+    if (!show) {
+      return (
+        <Table.Row>
+          <Table.Cell className="details-headings">Location</Table.Cell>
+          <Table.Cell>
+            {location || assetDetail.asset_location}
```

Comment:
Out of curiosity, the issue that necessitated the addition of the Logical OR was not fixed by our change to `assetDetail`?

Reply:
Yes, it's still necessary for the scenario where the asset detail page is your entry point to the app or when you reload. For some reason, the initial state of our component is not being set to the location we are getting from props